### DISCUSSION TITLE
Port Forwarding with a Custom Wireguard Connection

### DISCUSSION THREAD

**Discussion by johannesvr96** [qdm12/gluetun#839] [Q&A]
Hello, 

I have following problem:

 

I can't successfully forward Ports to my VPS with a static public IPV4 address (116.203.XXX.XXX) via GluTun VPN, Custom Provider.
But the container log itself is showing a successfull connection to the VPS.

 

My setup: Unraid Server (behind a Fritzbox home router, only with DS-Lite and no public IPV4 address). This server is running the GluetunVpn docker container and connects via Wireguard to the VPS successfully. The VPS (Ubuntu 20.04) has a Wireguard server installed and no firewall. Connection to the server is working. So I tried to setup port forwarding some container ports (like 8080 tcp for SABNZB to be reachable under the VPSServerIP:8080.

Interestingly, the port 8080 is then showing as open, when doing a PortScan on the 116.203.XXX.XXX:8080.

 

Every time: connection refused. UFW and other firewalls no set.

What my Port-Forwarding config of GlueTun looks like:
<img width="1294" alt="Screenshot UnRaidConfig" src="https://user-images.githubusercontent.com/53938166/152740036-db9c06c8-7d3c-49fd-81c7-0c2d1968fb5d.png">

 

IP's of my setup:

UnRaid Server: 192.168.178.21 (192.168.178.0/24 subnet)

VPS Server: Public IP 116.203.XXX.XXX, Wireguard IP: 10.66.66.1, Endpoint: 116.203.XXX.XXX, Interface eth0, wg0

GLUTUN Container: Wireguard IP 10.66.66.2

 

I also enabled IP Forwarding on the Linux VPS:

```
echo 'net.ipv4.ip_forward = 1' | sudo tee -a /etc/sysctl.conf
sudo sysctl -p /etc/sysctl.conf
```

 

What I've done wrong? I already asked the UnRaid Community, they couldn't help me.

Thank you very much

**Comment by qdm12** [comment]
Hello sorry for the delay.

Where is `SABNZB` running? On the vpn server or client side?

You can try with `FIREWALL=off` to check it works? That might leak data out of the VPN so don't use apart from testing.

**Comment by johannesvr96** [comment]
Hi, thanks for the answer.
SABNZB is running on the client side. Reason for this setup is, that the client has no public IP. 
I tried Firewall=off, but without any success.

Interestingly: If the client is a Windows Server, with SABNZB Ports open, I can access the WebInterface via the VPN-Server Public IP. So It has something to do with the client linux in any way...

**Reply by qdm12** [reply]
Sorry again for the huge delay... 

How do you connect sabnzb to gluetun? Are you using `network_mode: "container:gluetun"`?

You could also try the steps at https://github.com/qdm12/gluetun/wiki/VPN-server-port-forwarding#test-it

to check it works ruling out a sabnzb misconfiguration.

**Reply by johannesvr96** [reply]
Hi, I am using "--network=container:gluetun" as I am connecting SABNZB as a external container.

Thank you, will try that.

**Reply by johannesvr96** [reply]
> Sorry again for the huge delay...
> 
> How do you connect sabnzb to gluetun? Are you using `network_mode: "container:gluetun"`?
> 
> You could also try the steps at https://github.com/qdm12/gluetun/wiki/VPN-server-port-forwarding#test-it
> 
> to check it works ruling out a sabnzb misconfiguration.

Hi, I tried this setup again and had success accessing the "Port Test", it is showing the infomation Useragent etc.
So there must be problem between my SABNZB-Container and the Gluetun-Container. The Gluetun-Container has port forwarded the desired port.

**Reply by qdm12** [reply]
Perhaps ensure sabnzb listens on all network interfaces? (for example `:8080`)
Maybe it only listens on localhost or 0.0.0.0 (ipv4 only)
